plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -658,14 +659,31 @@ public Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession session
     }
 
     @Override
-    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode)
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode, boolean replace)
     {
         verify(transaction == null, "transaction already set");
-        transaction = newCreateTableTransaction(catalog, tableMetadata, session);
+        // TODO: Simplify this!
+        if (replace) {
+            IcebergTableHandle table = getTableHandle(session, tableMetadata.getTableSchema().getTable(), Optional.empty(), Optional.empty());
+            if (table != null) {
+                verifyTableVersionForUpdate(table);
+                Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
+                validateNotModifyingOldSnapshot(table, icebergTable);
+                getTableMetadata(session, getTableHandle(session, tableMetadata.getTableSchema().getTable(), Optional.empty(), Optional.empty()));
+                transaction = newCreateOrReplaceTableTransaction(catalog, tableMetadata, session);
```

Comment:
newCreateOrReplaceTableTransaction -> replaceTableTransaction ?
(here we  know it exists)

Reply:
I have refactored this slightly. to only have one transaction if replace (so it would be rather a createOrReplace transaction). I thought we are implementing `CREATE OR REPLACE` and not `REPLACE` so it seems more consistent to have that all the way reflected in the code.